vispy/gloo/texture.py

```diff
@@ -37,9 +37,11 @@ def convert_dtype_and_clip(data, dtype, copy=False):
         return new_data
 
 
-def downcast_to_32bit_if_needed(data, copy=False):
+def downcast_to_32bit_if_needed(data, copy=False, dtype=None):
```

Comment:
So what's the point of this new `dtype` kwarg? Looks like it is add the `data.astype(dtype)` functionality to this function, right? If so, OK.